src/test/scala/com/lunatech/pointingpoker/RoomSpec.scala

```diff
@@ -0,0 +1,204 @@
+package com.lunatech.pointingpoker
+
+import java.util.UUID
+
+import akka.actor.{Actor, ActorSystem}
+import akka.testkit.{TestActorRef, TestProbe}
+import com.lunatech.pointingpoker.websocket.WSMessage
+import com.lunatech.pointingpoker.websocket.WSMessage.MessageType
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.matchers.must
+import org.scalatest.wordspec.AnyWordSpec
+
+class RoomSpec extends AnyWordSpec with must.Matchers with BeforeAndAfterAll {
+  import RoomSpec._
+
+  implicit val system: ActorSystem = ActorSystem("RoomSpec")
+
+  override def afterAll(): Unit = {
+    system.terminate()
+  }
+
+  "Room Actor" should {
+    "update current issue and broadcast it" in {
+      val issue               = "Issue test 1"
+      val (user, userProbe)   = createUser(UUID.randomUUID(), "user1", false, "")
+      val (user2, user2Probe) = createUser(UUID.randomUUID(), "user2", false, "")
+      val actingUserId        = UUID.randomUUID()
+      val roomRef             = createRoom(UUID.randomUUID(), List(user, user2))
+      val expectedMessage =
+        WSMessage(MessageType.EditIssue, roomRef.underlyingActor.roomId, actingUserId, issue)
+
+      roomRef.underlyingActor.currentIssue mustBe ""
+
+      roomRef ! Room.EditIssue(actingUserId, issue)
+
+      roomRef.underlyingActor.currentIssue mustBe issue
+      roomRef.underlyingActor.issueLastEditBy mustBe Option(actingUserId)
+
+      userProbe.expectMsg(expectedMessage)
+      user2Probe.expectMsg(expectedMessage)
+    }
+
+    "clear votes and broadcast it" in {
+      val (user, userProbe)   = createUser(UUID.randomUUID(), "user1", true, "3")
+      val (user2, user2Probe) = createUser(UUID.randomUUID(), "user2", true, "5")
+      val actingUserId        = UUID.randomUUID()
+      val roomRef             = createRoom(UUID.randomUUID(), List(user, user2))
+      val expectedMessage = WSMessage(
+        MessageType.Clear,
+        roomRef.underlyingActor.roomId,
+        actingUserId,
+        WSMessage.NoExtra
+      )
+
+      roomRef ! Room.ClearVotes(actingUserId)
+
+      roomRef.underlyingActor.users.foreach { u =>
+        u.voted mustBe false
+        u.estimation mustBe ""
+      }
+      userProbe.expectMsg(expectedMessage)
+      user2Probe.expectMsg(expectedMessage)
+    }
+
+    "broadcast show votes" in {
+      val (user, userProbe)   = createUser(UUID.randomUUID(), "user1", true, "3")
+      val (user2, user2Probe) = createUser(UUID.randomUUID(), "user2", true, "5")
+      val actingUserId        = UUID.randomUUID()
+      val roomRef             = createRoom(UUID.randomUUID(), List(user, user2))
+      val expectedMessage =
+        WSMessage(MessageType.Show, roomRef.underlyingActor.roomId, actingUserId, WSMessage.NoExtra)
```

Comment:
This is sort of random, but since I noticed it here, why is the `currentIssue` represented in the message as just `extra`? I find that a bit confusing as it doesn't really tell me what this is at all here. Does the `extra` parameter ever represent anything other than the `currentIssue`? If not, I'd consider renaming it in `WSMessage`.

_Sort of_ related. What are you thoughts on changing the way we model `currentIssue`:

```diff
- var currentIssue: String = ""
+ var currentIssue: Option[String] = None
```

Currently I found there isn't a good way to differentiate between there being no issue or not unless we check to see if the string is empty or not around the code base. We don't really do that, but it might be a nicer way to model to this to see if there is a current issue set or not.

Reply:
WSMessage extra is also used to pass the estimation when voting, so WSMessage is suppose to be a part of WebSocket domain, so the extra can be anything need for that specific message type.
As for the model of currentIssue, right now it is just used to store the value, there isn't any rules that treat differently if the currentIssue is present or not, if that necessity comes along, I think it could be valid to re-think the model